tools/merge_note_registries.py

```
@@ -0,0 +1,137 @@
+
+
+def sweep_notes_dir(notes_dir: Path, url_registry: dict) -> dict:
+    """Scan notes_dir for index.md files whose original_url is absent from url_registry.
+
+    Returns a dict of {normalized_url: date_str} for URLs that were missing.
+    Does NOT mutate url_registry — caller decides whether to merge.
+    """
+    additions = {}
+    for index_md in sorted(notes_dir.rglob("index.md")):
+        try:
+            with open(index_md, "r", encoding="utf-8") as f:
```

Ruff (0.16.1)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)